src/provider/provider_coarse.c

```diff
@@ -411,14 +411,25 @@ static block_t *free_blocks_rm_node(struct ravl *free_blocks,
 // free_block_merge_with_prev - merge the given free block
 // with the previous one if both are unused and have continuous data.
 // Remove the merged block from the tree of free blocks.
-static block_t *free_block_merge_with_prev(struct ravl *all_blocks,
-                                           struct ravl *free_blocks,
-                                           block_t *block) {
+static block_t *free_block_merge_with_prev(
+    umf_memory_provider_handle_t upstream_memory_provider,
+    struct ravl *all_blocks, struct ravl *free_blocks, block_t *block) {
+
+    assert(all_blocks);
+    assert(free_blocks);
     assert(block);
 
+    // check if blocks could be merged by the upstream provider
+    enum umf_result_t merge_success = UMF_RESULT_ERROR_UNKNOWN;
     if (block->prev && block->prev->used == false &&
```

Comment:
I think that a check for the usage of `block` (`block->used == false`) is missing.

Reply:
OK